lib/sigsci-agent.sh

```diff
@@ -87,35 +87,43 @@ then
+  if [ -z $SIGSCI_AGENT_DOWNLOAD_URL ]
   then
-    SIGSCI_AGENT_VERSION=$(curl -s -L --retry 45 --retry-delay 2 https://dl.signalsciences.net/sigsci-agent/VERSION)
-  fi
-
-  # check if version exists.
-  STATUS=$(curl -s --retry 45 --retry-delay 2 -o /dev/null -w "%{http_code}" https://dl.signalsciences.net/sigsci-agent/${SIGSCI_AGENT_VERSION}/VERSION)
-  if [ $STATUS -ne 200 ]
-  then
-
-    # if we don't get a 200 response, skip agent installation.
-    (>&2 echo "-----> SigSci Agent version ${SIGSCI_AGENT_VERSION} not found or network unavailable after 90 seconds!")
-    (>&2 echo "-----> SIGSCI AGENT WILL NOT BE INSTALLED!")
+    # if agent version not specified then get the latest version.
+    if [ -z $SIGSCI_AGENT_VERSION ]
+    then
+      SIGSCI_AGENT_VERSION=$(curl -s -L --retry 45 --retry-delay 2 "https://dl.signalsciences.net/sigsci-agent/VERSION")
+    fi
 
-  else
-    echo "-----> Downloading sigsci-agent"
-    curl -s --retry 45 --retry-delay 2 -o sigsci-agent_${SIGSCI_AGENT_VERSION}.tar.gz https://dl.signalsciences.net/sigsci-agent/${SIGSCI_AGENT_VERSION}/linux/sigsci-agent_${SIGSCI_AGENT_VERSION}.tar.gz > /dev/null
+    # check if version exists.
+    STATUS=$(curl -s --retry 45 --retry-delay 2 -o /dev/null -w "%{http_code}" "https://dl.signalsciences.net/sigsci-agent/${SIGSCI_AGENT_VERSION}/VERSION")
 
-    if [ -z $SIGSCI_DISABLE_CHECKSUM_INTEGRITY_CHECK ]
+    if [ $STATUS -ne 200 ]
     then
-        curl -s --retry 45 --retry-delay 2 -o sigsci-agent_${SIGSCI_AGENT_VERSION}.tar.gz.sha256 https://dl.signalsciences.net/sigsci-agent/${SIGSCI_AGENT_VERSION}/linux/sigsci-agent_${SIGSCI_AGENT_VERSION}.tar.gz.sha256 > /dev/null
+      # if we don't get a 200 response, skip agent installation.
+      (>&2 echo "-----> SigSci Agent version ${SIGSCI_AGENT_VERSION} not found or network unavailable after 90 seconds!")
+      (>&2 echo "-----> SIGSCI AGENT WILL NOT BE INSTALLED!")
+    else
+      echo "-----> Downloading sigsci-agent"
+      curl -s --retry 45 --retry-delay 2 -o sigsci-agent_${SIGSCI_AGENT_VERSION}.tar.gz "https://dl.signalsciences.net/sigsci-agent/${SIGSCI_AGENT_VERSION}/linux/sigsci-agent_${SIGSCI_AGENT_VERSION}.tar.gz" > /dev/null
+
+      if [ -z $SIGSCI_DISABLE_CHECKSUM_INTEGRITY_CHECK ]
+      then
+        curl -s --retry 45 --retry-delay 2 -o sigsci-agent_${SIGSCI_AGENT_VERSION}.tar.gz.sha256 "https://dl.signalsciences.net/sigsci-agent/${SIGSCI_AGENT_VERSION}/linux/sigsci-agent_${SIGSCI_AGENT_VERSION}.tar.gz.sha256" > /dev/null
 
         # validate the gzip file
         if [[ "$(shasum -c sigsci-agent_${SIGSCI_AGENT_VERSION}.tar.gz.sha256)" != *"OK"* ]]
         then
             (>&2 echo "-----> sigsci-agent not installed because checksum integrity check failed")
             exit 1
         fi
+      fi
+    fi
+    else
+      # download the build pack from a local repo
+      echo "-----> Using local SigSci Agent repository ${SIGSCI_AGENT_DOWNLOAD_URL}"
```

Comment:
` Downloading SigSci Agent from `
is succinct; there's no guarantee that the url is "internal" or "external"; it's just an override.